components/02-elements/button/_button.scss

```diff
@@ -1,121 +1,116 @@
 
 .button {
-    display: inline-flex;
-    justify-content: center;
-    vertical-align: top;
+    display: inline-block;
     cursor: pointer;
-    height: $button-height;
-    min-width: $button-width;
-    margin: $button-margin;
-    padding: $button-padding;
-    border: $button-border;
-    outline: $button-outline;
-    background: $button-background;
-    font-family: $button-font-family;
-    font-size: $button-font-size;
-    color: $button-font-color;
-    transition: $transition-base;
-    align-items: center;
+    height: $button__height;
+    min-width: $button__width;
+    padding: $button__padding;
+    border: $button__border;
+    border-radius: $button__border-radius;
+    outline: $button__outline;
+    background: $button__background;
+    font-family: $button__font-family;
+    font-size: $button__font-size;
+    color: $button__font-color;
+    transition: $button__transition;
+    font-weight: $button__font-weight;
 
-    &:hover {
-        background: $button-hover-background;
-        border-color: $button-hover-border;
+    &:hover,
+    &:focus {
+        background: $button__background-hover;
+        border-color: $button__border-hover;
+        color: $button__font-color-hover;
     }
 
     &--link {
+        display: inline-flex;
         align-items: center;
-        &:hover {
-            text-decoration: none;
-            color: $button-font-color;
-        }
-    }
+        text-decoration: none;
 
-    &--native-link {
-        height: auto;
-        margin: 0;
-        padding: 0;
-        background: none;
-        border: none;
-        color: $link-color;
-        &:hover {
-            background: none;
-            color: $link-hover-color;
-            text-decoration: $link-hover-decoration;
+        &:hover,
+        &:focus {
+            text-decoration: none;
+            color: $button__font-color-hover;
         }
     }
 
     &--secondary {
-        border: $button-secondary-border;
-        color: $button-secondary-font-color;
-        background: $button-secondary-background;
+        border: $button__border--secondary;
+        color: $button__font-color--secondary;
+        background: $button__background--secondary;
 
-        &:hover {
-            border-color: $button-secondary-hover-border;
-            color: $button-secondary-hover-font-color;
-            background: $button-secondary-background;
+        &:hover,
+        &:focus {
+            border-color: $button__border-hover--secondary;
+            color: $button__font-color-hover--secondary;
+            background: $button__background-hover--secondary;
         }
     }
 
     &--icon {
-        padding: $button-icon-padding;
-        border: $button-icon-border;
-        background: $button-icon-background;
+        padding: $button-icon__padding;
+        border: $button-icon__border;
+        background: $button-icon__background;
+        transition: $button__transition;
```

Comment:
this transition is not necessary b/c it's herited from `.button` main element. You should add the transition to `.button__icon` element, please :)

Reply:
Thanks, done :)